### PR TITLE
Homework for third lecture.

### PR DESCRIPTION
**Pull request by Khazel**
Completed homework tasks for third lecture. Completed tasks 1-9, 11-12, to complete tasks 10, 13, 14 in additional commit.

### CODE REVIEW

File: .idea/.name
Comment (KaloyanTsvetkov) [inline .idea/.name:1]: Please add /.idea to your .gitignore file. This folder must not be part of the repo.

File: src/Lecture_3/Homework_3/Task_4.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public static void main(String[] args) {
+
+        int [] array = {23, 4, 4, 23};
+        int [] reversedArray = new int [array.length];
+
+        int length = array.length;
+        int start = 0;
+        int end = length - 1;
+
+        for (int i = 0; i < array.length; i++) {
```
Comment (KaloyanTsvetkov) [inline src/Lecture_3/Homework_3/Task_4.java]: You don't end extra array.
You could do the check by using only one loop.

File: src/Lecture_3/Homework_3/Task_10_Not_Completed.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        int [] array = {4, 16, 7, 8, 12, 23, 56, 64};
+        int start = 0;
+        int end = array.length -1;
+        int length = array.length;
+        boolean isZigZag = false;
+
+        if (array.length % 2 == 0) {
+            for (int i = start; i < length; i++) {
+                if (i % 3 == 0) {
+                    // check that first/previous element is smaller than next, and then check that second is bigger than
```
Comment (KaloyanTsvetkov) [inline src/Lecture_3/Homework_3/Task_10_Not_Completed.java]: You have the solution as part of the cod efor the next lecture

File: src/Lecture_3/Homework_3/Task_13.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+public class Task_13 {
+
+    public static void main(String[] args) {
+
+        Scanner sn = new Scanner(System.in);
+
+        System.out.println("Please enter the number of rows to form square two dimensional array");
+        int rows = sn.nextInt();
+        System.out.println("Please enter the number of columns to form square two dimensional array");
```
Comment (KaloyanTsvetkov) [inline src/Lecture_3/Homework_3/Task_13.java]: In order to have diagonals your array must be square. You need only 1 user input, not 2.

File: src/Lecture_3/Homework_3/Task_13.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            if (rows != columns) {
+                System.out.println("Your array is not square.");
+            } else {
+                int [][] array = new int [rows][columns];
+
+                System.out.println("Please enter elements of array");
+
+                for (int i = 0; i < array.length; i++) {
+                    for (int j = 0; j < array[0].length; j++) {
+                        array[i][j] = sn.nextInt();
```
Comment (KaloyanTsvetkov) [inline src/Lecture_3/Homework_3/Task_13.java]: Main diagonal is: arr[i][i]

**Review by KaloyanTsvetkov — APPROVED** [review]

File: .idea/.name
Comment (Khazel) [reply .idea/.name:1]: Done in pull request for homework for fourth lecture.

File: src/Lecture_3/Homework_3/Task_4.java (hunk 1 above)
Comment (Khazel) [reply src/Lecture_3/Homework_3/Task_4.java]: Got it, changed code to not to use second array.

File: src/Lecture_3/Homework_3/Task_10_Not_Completed.java (hunk 2 above)
Comment (Khazel) [reply src/Lecture_3/Homework_3/Task_10_Not_Completed.java]: Got it, just wanted to try to resolve it myself.

File: src/Lecture_3/Homework_3/Task_13.java (hunk 3 above)
Comment (Khazel) [reply src/Lecture_3/Homework_3/Task_13.java]: Thanks, that makes sense. Corrected code to accept only one input, excluded validation for equal raws and columns values.

File: src/Lecture_3/Homework_3/Task_13.java (hunk 4 above)
Comment (Khazel) [reply src/Lecture_3/Homework_3/Task_13.java]: Sorry, I didn't understand this comment.

**Pull request merged**